libs/utils/utils.js

```diff
@@ -245,14 +245,20 @@ export const loadScript = (url, type) => new Promise((resolve, reject) => {
     if (type) {
       script.setAttribute('type', type);
     }
-    script.onload = () => {
-      resolve(script);
-    };
-    script.onerror = () => {
-      reject(new Error('error loading script'));
-    };
     head.append(script);
   }
+
+  if (script.dataset.loaded) {
+    resolve(script);
+  }
+
+  script.addEventListener('load', () => {
+    script.dataset.loaded = true;
+    resolve(script);
+  });
```

Comment:
minor: clean up event listeners
```suggestion
  const onLoad = () => {
    script.dataset.loaded = true;
    script.removeEventListener('load', onLoad);
    script.removeEventListener('error', onError);
    resolve(script);
  }
  script.addEventListener('load', onLoad);
```

Reply:
This is something I hate about event listeners, you can't reference both onLoad and onError from both function. 
I'll have to make this one function and use `event.type`.